person-service/src/main/kotlin/com/thehuginn/messaging/CreateAddressCommandConsumer.kt

```diff
@@ -0,0 +1,40 @@
+package com.thehuginn.messaging
+
+import com.thehuginn.domain.Person
+import com.thehuginn.exception.NotFoundException
+import com.thehuginn.messaging.dto.command.CreateAddressCommandMessage
+import com.thehuginn.repository.MessageRepository.Message.CREATE_ADDRESS
+import com.thehuginn.service.AddressService
+import com.thehuginn.service.PersonService
+import com.thehuginn.service.command.mapper.AddressCommandMapper
+import io.smallrye.mutiny.Uni
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.transaction.Transactional
+import org.eclipse.microprofile.reactive.messaging.Incoming
+import java.util.logging.Logger
+
+@ApplicationScoped
+class CreateAddressCommandConsumer (
+    private val personService: PersonService,
+    private val addressService: AddressService,
+    private val addressCommandMapper: AddressCommandMapper
+) {
+
+    private val logger = Logger.getLogger(CreateAddressCommandConsumer::class.java.toString())
+
+    @Incoming(CREATE_ADDRESS)
+    @Transactional
+    fun process(message: CreateAddressCommandMessage): Uni<Void> =
+        Uni.createFrom().item(message)
+            .onItem().transformToUni { msg ->
+                Uni.createFrom().item { personService.getPerson(msg.person_id) }
+                    .onItem().ifNull().failWith(NotFoundException(Person::class.java, msg.person_id))
+                    .replaceWith(msg)
+            }
+            .onItem().transformToUni { msg ->
+                Uni.createFrom().item(addressService.persist(addressCommandMapper.mapFrom(msg)))
```

Comment:
taktiez, tu by malo byt to iste. Resp tu staci iba .item(addressCommandMapper.mapFrom(msg).onItem().call { addressService }